agent.go

```diff
@@ -156,6 +156,22 @@ var collatedTrace = false
 // if true, coredump when an internal error occurs or a fatal signal is received
 var crashOnError = false
 
+// commType is used to denote the communication channel type used.
+type commType int
+
+const (
+	// virtio-serial channel
+	serialCh commType = iota
+
+	// vsock channel
+	vsockCh
```

Comment:
Because there's a new line and comments between `serialCh` and `vsockCh`, and `unknownCh`, I think both `vsockCh` and `unknownCh` are going to be typed as `int` instead of `commType`.

Reply:
@sboeuf I was able to assign between those values, I think compile would have complained in that case.

Reply:
You can still compile since it's a cast later in the code, but unless the Go compiler got smarter, this was a problem I had encountered before.

Reply:
No golang will never cast automatically.
Looks like newline are allowed according to this wiki: https://github.com/golang/go/wiki/Iota